src/main/java/gyro/google/compute/RouteResource.java

```diff
@@ -0,0 +1,345 @@
+/*
+ * Copyright 2019, Perfect Sense, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package gyro.google.compute;
+
+import com.google.api.client.googleapis.json.GoogleJsonResponseException;
+import com.google.api.services.compute.Compute;
+import com.google.api.services.compute.model.Operation;
+import com.google.api.services.compute.model.Route;
+import com.google.cloud.compute.v1.ProjectGlobalNetworkName;
+import gyro.core.GyroException;
+import gyro.core.GyroUI;
+import gyro.core.Type;
+import gyro.core.resource.Output;
+import gyro.core.resource.Resource;
+import gyro.core.scope.State;
+import gyro.core.validation.Range;
+import gyro.core.validation.Required;
+import gyro.core.validation.ValidationError;
+import gyro.google.Copyable;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Objects;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+/**
+ * Creates a route.
+ *
+ * Example
+ * -------
+ *
+ * .. code-block:: gyro
+ *
+ *     google::route route-example
+ *         name: "route-example"
+ *         description: "route-example-description"
+ *         network: $(google::network network-example-route)
+ *         next-hop-ip: "10.0.0.5"
+ *         dest-range: "0.0.0.0/0"
+ *     end
+ */
+@Type("route")
+public class RouteResource extends ComputeResource implements Copyable<Route> {
+    private String name;
+    private String description;
+    private String destRange;
+    private NetworkResource network;
+    private Long priority;
+    private Set<String> tags;
+    private String nextHopGateway;
+    private String nextHopVpnTunnel;
+    private InstanceResource nextHopInstance;
+    private String nextHopIp;
+
+    // Read-only
+    private String id;
+    private String selfLink;
+    private List<RouteWarning> warnings;
```

Comment:
I think these warnings should be output after a create or update rather than stored.